packages/@react-aria/selection/src/useSelectableItem.ts

```diff
@@ -135,24 +135,24 @@ export function useSelectableItem(options: SelectableItemOptions): SelectableIte
   };
 
   // Focus the associated DOM node when this item becomes the focusedKey
-  let isFocused = key === manager.focusedKey;
   useEffect(() => {
+    let isFocused = key === manager.focusedKey;
     if (isFocused && manager.isFocused && !shouldUseVirtualFocus && document.activeElement !== ref.current) {
       if (focus) {
         focus();
       } else {
         focusSafely(ref.current);
       }
     }
-  }, [ref, isFocused, manager.focusedKey, manager.childFocusStrategy, manager.isFocused, shouldUseVirtualFocus]);
+  }, [ref, key, manager.focusedKey, manager.childFocusStrategy, manager.isFocused, shouldUseVirtualFocus]);
 
   // Set tabIndex to 0 if the element is focused, or -1 otherwise so that only the last focused
   // item is tabbable.  If using virtual focus, don't set a tabIndex at all so that VoiceOver
   // on iOS 14 doesn't try to move real DOM focus to the item anyway.
   let itemProps: SelectableItemAria['itemProps'] = {};
   if (!shouldUseVirtualFocus) {
     itemProps = {
-      tabIndex: isFocused ? 0 : -1,
+      tabIndex: key === manager.focusedKey ? 0 : -1,
```

Comment:
should this be in state if render was too early to be accurately reflected?

Reply:
talked w/ @snowystinger and we think this will be ok. Here is the run down of the original problem and why we think the above tabIndex calculation should be ok.

Original problem:
If the destination list had a previously `focusedKey` tracked, the following happens when a user drags and drops a new item into the list:
Initial setup: source ListView and destination ListView each have one item each, we'll call them `sourceKey` and `destKey`. Destination ListView's `focusedKey` is set to `destKey` since the user previously had clicked/interacted with that row 

1. When the new item (`sourceKey`) is dropped and inserted into the destination list, `useSelectableItem`'s `isFocused` is calculated in render. At this moment `manager.focusedKey` for the destination ListView is still equal to the original `focusedKey` (aka `destKey`) and thus `isFocused` is true for the `destKey` row but is false for the `sourceKey` row. 
2. `useDroppableCollection`'s [useLayoutEffect](https://github.com/adobe/react-spectrum/blob/e8a1bf5b357d4f655e9927c27ec1b2490664e2ec/packages/%40react-aria/dnd/src/useDroppableCollection.ts#L188) triggers, setting the newly added `sourceKey` as the new `manager.focusedKey`. This change is reflected immediately in `manager.focusedKey` because that getter is tied to a [ref](https://github.com/adobe/react-spectrum/blob/e8a1bf5b357d4f655e9927c27ec1b2490664e2ec/packages/%40react-stately/selection/src/useMultipleSelectionState.ts#L100-L102). Since this is still the same render cycle as in step 1, `useSelectableItem`'s `isFocused` doesn't get recalculated (but will in the next render cycle) and is thus out of date 
3. `useSelectableItem`'s useEffect fires. The `destKey` row will then call `focus()` on itself since its `isFocused` is true as per step 1. 
4. The focus call triggers another [manager.setFocusedKey call](https://github.com/adobe/react-spectrum/blob/e8a1bf5b357d4f655e9927c27ec1b2490664e2ec/packages/%40react-aria/selection/src/useSelectableItem.ts#L156-L159) that will get batched with step 2's setFocusedKey call, essentially overwriting for the new render
5. The destination ListView rerenders, and the end result is that `manager.focusedKey` is still set to `destKey` instead of the recently added `sourceKey` row

With the change made in this PR to move `isFocused`'s calculation into the useEffect, we can calculate what key should be focused after the update to `manager.focusedKey` happens from `useDroppableColelction`'s useLayoutEffect and thus have the recently dropped + inserted row be focused. This then means the tabIndex calculation above should be correct since `manager.focusedKey` won't accidentally be overwritten through a side effect and thus the final render should be using the "correct" manager.focusedKey